src/edu/csus/ecs/pc2/ui/RunsTablePane.java

```diff
@@ -2106,31 +2179,36 @@ private void showSourceForSelectedRun() {
                             
                             //we got some RunFiles from the server; put them into the contest model
                             getContest().updateRunFiles(run, fetchedRunFiles);
-                            
                         } else {
                             
                             //we got a reply from the server but we didn't get any RunFiles
                             getController().getLog().log(Log.WARNING, "Server failed to return RunFiles in response to fetch run request");
-                            getController().getLog().log(Log.WARNING, "Unable to fetch source files for run " + run.getNumber() + " from server");
-                            showMessage("Unable to fetch selected run; check log");
-                            return;
+                            bFetchError = true;
                         }
                         
                     } else {
                         
                         // the server failed to reply to the fetchRun request within the time limit
                         getController().getLog().log(Log.WARNING, "No response from server to fetch run request after " + waitedMS + "ms");
-                        getController().getLog().log(Log.WARNING, "Unable to fetch run " + run.getNumber() + " from server");
-                        showMessage("Unable to fetch selected run; check log");
-                        return;
+                        bFetchError = true;
                     }
                 }
                 
+                // OK to now start another view source
+                AllowViewSource();
```

Comment:
I think there may be a potential synchronization error here (although I could probably be convinced otherwise).  See my comments on the subsequent call to `AllViewSource()`, below, around (new) line 2288 for details.